garden_ai/client.py

```diff
@@ -311,7 +311,11 @@ def _update_datacite(
         # "publish" in the event field moves the DOI from draft state to findable state
         # https://support.datacite.org/docs/how-do-i-make-a-findable-doi-with-the-rest-api
         if register_doi:
-            metadata.update(event="publish", url=f"https://thegardens.ai/{obj.doi}")
+            doi_split = obj.doi.split("/")
+            metadata.update(
```

Comment:
This is kind of nitpicky because I'm fairly certain that 100% of our DOIs will be minted following the 10.1234/abc-def convention, but I think the spec technically allows for arbitrarily many `/` in the suffix. The builtin `urllib.parse.quote` is a safer bet here to encode the doi string (and is generally very handy / good to know about)